### PR TITLE
feature: delete provider api key

### LINKED ISSUE #238: Ability to delete provider API keys
### Proposal

Currently, when we render API keys, we have a button to edit each API key. As #208 is in development, we'll need a way to allow customers to delete their API keys.

Simply add a red button with a trash icon (same as done in other places in the platform) that deletes an API key. Implement the API endpoint if it does not exist.

# Context
- Need to do some NestJS work (add a resolver mutation - `deleteProviderApiKey`, the service method already exists)
- React (add a button, GraphQL mutation definition + hook, call the resolver mutation)

### Use-Case

_No response_

### Is this a feature you are interested in implementing yourself?

Happy for interested community members to pick this up!

### PR DESCRIPTION
**Pull request by adityapuri75**
Fixes #238

### CODE REVIEW

**Comment by arielweinberger** [comment]
Can you please attach a screenshot for UI-related PRs? This way I don't have to run the code to see exactly what it looks like.

File: apps/console/src/app/components/api-keys/ProviderApiKeyListItem.tsx [hunk 1]
```diff
@@ -53,6 +64,16 @@ export const ProviderApiKeyListItem = ({
     },
   });
 
+  const handleDeleteProvider = async (provider: string) => {
+    await deleteProviderApiKey({ provider, organizationId: currentOrgId });
+    messageApi.open({
+      type: "success",
+      content: "Provider api key deleted",
```
Comment (arielweinberger) [inline apps/console/src/app/components/api-keys/ProviderApiKeyListItem.tsx:71]:
```suggestion
      content: "Provider API key was deleted successfully",
```

File: apps/console/src/app/components/api-keys/ProviderApiKeyListItem.tsx [hunk 2]
```diff
@@ -77,81 +98,101 @@ export const ProviderApiKeyListItem = ({
...
-                <Input
-                  placeholder="Paste your API key"
-                  autoComplete="off"
-                  style={{ marginTop: 22 }}
-                />
-              </Form.Item>
-            ) : (
-              <Typography.Text style={{ marginLeft: 10, opacity: 0.5 }}>
-                {value || "No API key provided"}
+        <p>Are you sure you want to delete the provider API key?</p>
```
Comment (arielweinberger) [inline apps/console/src/app/components/api-keys/ProviderApiKeyListItem.tsx:110]:
```suggestion
        <p>Are you sure you want to delete this API key?</p>
```

**Comment by arielweinberger** [comment]
I quickly cloned it and tested it. Looks good. Next time, please attach a screenshot for UI-related PRs? This way I don't have to run the code.

Well done @adityapuri75!

**Comment by adityapuri75** [comment]
Sure will add screenshot for UI related PRs, Thank you @arielweinberger 😀

**Pull request merged**